core/dbt/adapters/base/impl.py

```diff
@@ -1298,6 +1310,79 @@ def render_column_constraint(cls, constraint: ColumnLevelConstraint) -> str:
         else:
             return ""
 
+    @available
+    @classmethod
+    def render_raw_columns_constraints(cls, raw_columns: Dict[str, Any]) -> List:
+        rendered_column_constraints = []
+
+        for _, v in raw_columns.items():
+            rendered_column_constraint = [f"{v['name']} {v['data_type']}"]
+            for con in v["constraints"]:
+                constraint = cls._parse_column_constraint(con)
+                c = cls.process_parsed_constraint(constraint, cls.render_raw_column_constraint)
+                if c is not None:
+                    rendered_column_constraint.append(c)
+            rendered_column_constraints.append(" ".join(rendered_column_constraint))
+
+        return rendered_column_constraints
+
+    @classmethod
+    def process_parsed_constraint(
+        cls, parsed_constraint: Union[ColumnLevelConstraint, ModelLevelConstraint], render_func
```

Comment:
This implementation makes sense to me if we want to tie column-level and model-level support together, but I question whether we do. For example, Spark supports check constraints, but only at the model/table level, and of course not-null only makes sense at the column level anyway. What do you think @MichelleArk ?

Reply:
I wouldn't want to be too strict about this. As a general rule, on most data platforms, every constraint should be supported at both the model-level (multiple columns) and the single column-level. Of all the constraint types, `not_null` is the only one that is supported _only_ for a single column, and so it only really makes sense to define it at the column-level. IMO the funkiness of `check` on Spark/Databricks is more of a detail of their current implementation, which they're working to evolve/improve.

Reply:
To summarize - I think it is totally legit to use this same method for both column-level and model-level constraints